### PR TITLE
15051 Updated local filing (CTCO) to respond to UX Observations

### PR DESCRIPTION
**Pull request by JazzarKarim**
*Issue #:* /bcgov/entity#15051

*Description of changes:*
- Yui's Comment: https://github.com/bcgov/entity/issues/15051#issuecomment-1518335264
- Short Demo:

https://user-images.githubusercontent.com/122301442/234367862-77d46342-cf69-4ad3-a50e-5088efaae884.mp4

CC: @yuisotozaki

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/views/ConsentContinuationOut.vue [hunk 1]
```diff
@@ -91,16 +93,19 @@
...
-                  contactLabel="Business Office"
-                  :documentOptionalEmail="documentOptionalEmail"
-                  @update:optionalEmail="documentOptionalEmail=$event"
-                  @valid="documentDeliveryValid=$event"
-                />
-              </v-card>
+              <div :class="{ 'invalid-section': !documentDeliveryValid && showErrors }" id="document-delivery">
+                <v-card flat class="py-8 px-5">
+                  <DocumentDelivery
+                    id="document-delivery"
```
[the remark this answers is not in the document]
Comment (severinbeauvais) [reply src/views/ConsentContinuationOut.vue:99]: Same on line 119.

File: src/views/ConsentContinuationOut.vue [hunk 2]
```diff
@@ -147,6 +158,7 @@
               :offset="{ top: 120, bottom: 40 }"
             >
               <SbcFeeSummary
+                class="ml-4"
```
Comment (severinbeauvais) [inline src/views/ConsentContinuationOut.vue:161]: What do the other local filings (AR, COA, COD, Correction) do here?

What does this look like on mobile (responsive / narrow screen)? Normally the Affix component handles this. You may need a special ml class that only takes affect on sm breakpoints or larger. Ref: https://vuetifyjs.com/en/styles/spacing/

File: src/views/ConsentContinuationOut.vue [hunk 3]
```diff
@@ -553,6 +566,13 @@ export default class ConsentContinuationOut extends Vue {
    * or when user submits from Staff Payment dialog.
    */
   async onClickFilePay (fromStaffPayment = false): Promise<void> {
+    // if there is an invalid component, scroll to it
+    if (!this.isPageValid) {
+      this.showErrors = true
+      this.scrollToInvalidComponent()
+      return
+    }
```
Comment (severinbeauvais) [inline src/views/ConsentContinuationOut.vue:574]: Very nice :)

File: src/views/ConsentContinuationOut.vue [hunk 4]
```diff
@@ -812,6 +832,30 @@ export default class ConsentContinuationOut extends Vue {
...
+      certifyForm: this.certifyFormValid,
+      courtOrder: this.courtOrderValid
+    }
+  }
+
+  private async scrollToInvalidComponent (): Promise<void> {
+    // scroll to invalid components
+    await Vue.nextTick()
+    await this.validateAndScroll(this.validFlags, this.validComponents)
+  }
```
Comment (severinbeauvais) [inline src/views/ConsentContinuationOut.vue:857]: Why have this as a separate method?

**Review by severinbeauvais — COMMENTED** [review]
Could you add or  update some unit tests, please?

File: src/views/ConsentContinuationOut.vue (hunk 2 above)
Comment (JazzarKarim) [reply src/views/ConsentContinuationOut.vue]: The problem was that I don't know why, but I added no gutters to v-row. Removing it fixed the layout and now is consistent with other local filings.

**Comment by JazzarKarim** [comment]
> Could you add or update some unit tests, please?

Let me know if I should add more tests or fix any of the previously added ones!

File: tests/unit/ConsentContinuationOut.spec.ts [hunk 5]
```diff
@@ -8,17 +8,57 @@ import { ConfirmDialog, PaymentErrorDialog, ResumeErrorDialog, SaveErrorDialog,
...
 
 const store = getVuexStore() as any // remove typings for unit tests
 const localVue = createLocalVue()
 localVue.use(VueRouter)
 
 describe('Consent to Continuation Out view', () => {
+  let sinonAxiosGet: any
+
   beforeEach(() => {
+    sinonAxiosGet = sinon.stub(axios, 'get')
```
Comment (severinbeauvais) [inline tests/unit/ConsentContinuationOut.spec.ts:34]: If you just have 1 mock getter then you don't need to assign it to a variable -- just just it directly on line 39.

(see my next comment)

File: tests/unit/ConsentContinuationOut.spec.ts [hunk 6]
```diff
@@ -8,17 +8,57 @@ import { ConfirmDialog, PaymentErrorDialog, ResumeErrorDialog, SaveErrorDialog,
...
+
+    store.commit('setTestConfiguration', { key: 'VUE_APP_PAY_API_URL', value: 'https://pay.web.url/' })
+
+    // mock "get orig filing" endpoint
+    sinonAxiosGet
+      .withArgs('businesses/CP1234567/filings/123')
+      .returns(new Promise(resolve =>
```
Comment (severinbeauvais) [inline tests/unit/ConsentContinuationOut.spec.ts:41]: This is the "old way" of mocking a get call.

The new way is to just mock the LegalServices call. After all, you don't need to test Axios (and it's cleaner, too). Example: https://github.com/bcgov/business-filings-ui/blob/0cae8df3d15f5f5c6510d6381d611ee3b006ed39/tests/unit/BusinessActions.spec.ts#L20

**Review by severinbeauvais — APPROVED** [review]
Fantastic work! Just get more approvals and remember to update the app version.

**Review by ketaki-deodhar — APPROVED** [review]
Looks good! 👍

File: tests/unit/ConsentContinuationOut.spec.ts (hunk 6 above)
Comment (JazzarKarim) [reply tests/unit/ConsentContinuationOut.spec.ts]: Alright, noted! I looked also at various other examples in the project. I'll be keeping this in mind.

**Comment by JazzarKarim** [comment]
> Fantastic work! Just get more approvals and remember to update the app version.

Thank you so much Sev!

File: tests/unit/ConsentContinuationOut.spec.ts (hunk 6 above)
Comment (severinbeauvais) [reply tests/unit/ConsentContinuationOut.spec.ts]: This is the oldest UI so you'll find some older ways of doing things 😁

Comment (JazzarKarim) [reply tests/unit/ConsentContinuationOut.spec.ts]: I managed to update my unit test to use the new way. I like it 🕺

Comment (severinbeauvais) [reply tests/unit/ConsentContinuationOut.spec.ts]: Beauty.

**Pull request merged**